lib/logstash/inputs/beats/message_listener.rb

```diff
@@ -31,7 +31,12 @@ def onNewMessage(ctx, message)
       hash = message.getData
       ip_address = ip_address(ctx)
 
-      hash['@metadata']['ip_address'] = ip_address unless ip_address.nil? || hash['@metadata'].nil?
+      unless ip_address.nil? || hash['@metadata'].nil?
+        tmp_evt = org.logstash.Event.new
+        tmp_evt.setField(@input.field_hostip, ip_address)
+        # Leverage Logstash's event to convert nested event fields into nested hashes
+        nested_fields_to_maps(hash, tmp_evt)
```

Comment:
:thinking: - turned out to be a bit complex, but I guess it does what it needs to do

Reply:
I think the same, but the alternative was to replicate to logic to unpack nested fields names into maps with essentially replicating the behavior inside the plugin, that was even worst since we introduced duplication.